rules/CodeQuality/Rector/Property/OrderByKeyToClassConstRector.php

```diff
@@ -0,0 +1,114 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Doctrine\CodeQuality\Rector\Property;
+
+use Doctrine\Common\Collections\Criteria;
+use PhpParser\Node;
+use PhpParser\Node\Expr\Array_;
+use PhpParser\Node\Scalar\String_;
+use PhpParser\Node\Stmt\Property;
+use Rector\BetterPhpDocParser\PhpDoc\DoctrineAnnotationTagValueNode;
+use Rector\Core\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Doctrine\Tests\CodeQuality\Rector\Property\OrderByKeyToClassConstRector\OrderByKeyToClassConstRectorTest
+ */
+final class OrderByKeyToClassConstRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition(
+            'Replace OrderBy Attribute ASC/DESC with class constant from Criteria',
+            [
+                new CodeSample(
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+class ReplaceOrderByAscWithClassConstant
+{
+    #[ORM\OrderBy(['createdAt' => 'ASC'])]
+    protected \DateTimeInterface $messages;
+}
+?>
+CODE_SAMPLE
+
+                    ,
+                    <<<'CODE_SAMPLE'
+use Doctrine\ORM\Mapping as ORM;
+
+class ReplaceOrderByAscWithClassConstant
+{
+    #[ORM\OrderBy(['createdAt' => \Doctrine\Common\Collections\Criteria::ASC])]
+    protected \DateTimeInterface $messages;
+}
+?>
+CODE_SAMPLE
+                ),
+            ]
+        );
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Property::class];
+    }
+
+    /**
+     * @param Property $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $nodeAttribute = null;
+        foreach ($node->attrGroups as $attrGroup) {
+            foreach ($attrGroup->attrs as $attr) {
+                if ($attr->name->toString() === 'Doctrine\ORM\Mapping\OrderBy') {
+                    $nodeAttribute = $attr;
```

Comment:
I guess orderby setting will be only once, so you can stop loop both outer and inner loop with break 2;

```
break 2;
```

Reply:
Your right good point!